### PR TITLE
chore: Compatibility with dbplyr > 2.3.2

### PR DESCRIPTION
**Pull request by mgirlich**
This makes dm compatible with https://github.com/tidyverse/dbplyr/pull/1284.

* use `DBI::Id()` consistently instead of `ident_q()` etc.
* update to new interface of `sql_query_append()`
* Switch to `remote_table()` for new dbplyr (`remote_name()` now only returns the unqualified name of the table)

### CODE REVIEW

File: R/dm_from_con.R [hunk 1]
```diff
@@ -139,15 +134,9 @@ quote_ids <- function(x, con, schema = NULL) {
   }
 
   if (is_null(schema)) {
-    map(
-      x,
-      ~ dbplyr::ident_q(dbplyr::build_sql(dbplyr::ident(.x), con = con))
-    )
+    map_if(x, ~ !inherits(.x, "Id"), ~ DBI::Id(table = .x))
```
Comment (mgirlich) [inline R/dm_from_con.R:137]: This feels more like a hack but I just wanted to get this working.

File: R/learn.R [hunk 2]
```diff
@@ -82,7 +82,8 @@ dm_learn_from_db <- function(dest, dbname = NA, schema = NULL, name_format = "{t
     ungroup() %>%
     pull_tbl()
 
-  tables <- map2(table_info$from, table_info$vars, ~ tbl(con, dbplyr::ident_q(.x), vars = .y))
+  table_info$from_id <- DBI::dbUnquoteIdentifier(con, DBI::SQL(table_info$from))
```
Comment (mgirlich) [inline R/learn.R:85]: We need to use `dbUnquoteIdentifier()` because `group_by()` doesn't work with S4 classes.

File: R/learn.R [hunk 3]
```diff
@@ -173,19 +174,24 @@ dm_learn_from_db <- function(dest, dbname = NA, schema = NULL, name_format = "{t
...
 schema_if <- function(schema, table, con, dbname = NULL) {
-  table_sql <- DBI::dbQuoteIdentifier(con, table)
   if (is_null(dbname) || is.na(dbname) || dbname == "") {
-    if_else(
-      are_na(schema),
-      table_sql,
-      # need 'coalesce()' cause in case 'schema' is NA, 'if_else()' also tests
-      # the FALSE option (to see if same class) and then 'dbQuoteIdentifier()' throws an error
-      SQL(paste0(DBI::dbQuoteIdentifier(con, coalesce(schema, "")), ".", table_sql))
+    purrr::map2(
```
Comment (mgirlich) [inline R/learn.R:178]: This would be nicer with the new table ident class...

File: R/mssql.R [hunk 4]
```diff
@@ -6,6 +6,7 @@ mssql_sys_db <- function(con, dbname, name, vars = NULL) {
     fq_name <- paste0(dbname, ".", name)
     sql_name <- dbname
   }
+  browser()
```
Comment (mgirlich) [inline R/mssql.R:9]: I didn't do any changes to MS SQL as I don't have a database to test this with. I left this `browser()` as a reminder 😄

File: R/rows-dm.R [hunk 5]
```diff
@@ -285,17 +285,27 @@ do_rows_append <- function(x, y, by = NULL, ..., in_place = FALSE, autoinc_col =
...
   con <- dbplyr::remote_con(x)
-  target_name <- dbplyr::remote_name(x)
-  insert_queries <- map(source_rows, ~ dbplyr::sql_query_append(
-    con,
-    target_name,
-    .x,
-    returning_cols = autoinc_col
-  ))
+  if (utils::packageVersion("dbplyr") >= "2.3.2.9000") {
+    insert_queries <- map(source_rows, ~ dbplyr::sql_query_append(
```
Comment (mgirlich) [inline R/rows-dm.R:289]: `sql_query_append()` etc got an updated interface to make them more consistent to other `sql_query_*()` functions. In particular, it felt wrong to pass a lazy table to them.

**Comment by krlmlr** [comment]
Thanks. Working on a Docker Compose file (plus instructions) to help everybody run the necessary databases. Would that help you?

**Comment by krlmlr** [comment]
✅ ✅ ✅ 

🎉 🎉 🎉 🎉 🎉 

Now ready to actually review the code. Do you expect this also to work with the CRAN version of dbplyr?

**Comment by mgirlich** [comment]
I think you already answered the question yourself 😄 Glad to see that not so many changes were needed.

**Comment by krlmlr** [comment]
Thanks a lot for your help!

**Pull request merged**